packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_icx_create_order.test.ts

```diff
@@ -0,0 +1,146 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { getProviders, MockProviders } from '../provider.mock'
+import { P2WPKHTransactionBuilder } from '../../src'
+import { calculateTxid, fundEllipticPair, sendTransaction } from '../test.utils'
+import BigNumber from 'bignumber.js'
+import { OP_CODES, ICXCreateOrder, ICXOrderType } from '@defichain/jellyfish-transaction'
+
+describe('create ICX order', () => {
+  const container = new MasterNodeRegTestContainer()
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+
+    providers = await getProviders(container)
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  beforeEach(async () => {
+    await providers.randomizeEllipticPair()
+    await container.waitForWalletBalanceGTE(11)
+    await fundEllipticPair(container, providers.ellipticPair, 10)
+    await providers.setupMocks()
+    await fundEllipticPair(container, providers.ellipticPair, 20)
+  })
+
+  it('should create ICX order', async () => {
+    const script = await providers.elliptic.script()
+    const icxOrder: ICXCreateOrder = {
+      orderType: ICXOrderType.EXTERNAL,
```

Comment:
For external order type `receivePubkey` is optional. it does not use `receivePubkey` inside defid.
Btw c++ rpc description needs to be updated too.

Reply:
Second test is without `receivePubkey`